src/file-events/cpp/generic_fsnotifier.cpp

```diff
@@ -163,9 +164,55 @@ void AbstractServer::reportChange(JNIEnv* env, int type, const u16string& path)
     jstring javaPath = env->NewString((jchar*) path.c_str(), (jsize) path.length());
     env->CallVoidMethod(watcherCallback, watcherCallbackMethod, type, javaPath);
     env->DeleteLocalRef(javaPath);
+
+    jthrowable exception = env->ExceptionOccurred();
+    if (exception != nullptr) {
+        env->ExceptionDescribe();
```

Comment:
Shouldn't we better hand the exception to the logging instead of printing the stacktrace to stderr?

Reply:
We could, though we don't currently have a way to log exceptions from the native side. Note that we are currently passing an already formatted string to logging, and here we'd need to pass a `jobject`.

I think for now this will do.

Reply:
My fear is that the logging will be lost instead of being part of the daemon log. We saw this when printing to stdout as well. Maybe it will work since it uses Java code to print of the stacktrace?

Reply:
Actually, I can do this in `NativeFileWatcherCallback` already...